### PR TITLE
[Feature] Automatic profile regeneration

### PR DESCRIPTION
**Pull request by omarzl**
This pr adds the feature proposed here https://github.com/Tinder/sign-here/issues/21

What changes:
- If the argument profileName exists, a request to the API will consult if there is a profile with that name.
- In case the profile exists, the flag `--auto-regenerate` is set and the number of devices changed, the profile will be deleted and created again with all the device ids.
- In case the profile exists but the flag `--auto-regenerate` isn't set, then the profile will be saved to the `outputPath` and the process will exit successfully.
- Since this new flag needs the profileName, a new error `profileNameMissing` will be triggered when the profile name doesn't exist.
- The logic in the class `iTunesConnectServiceImp` that determines if the profile should register devices has been moved to the enum `ProfileType` so it can be reused in `CreateProvisioningProfileCommand`

### CODE REVIEW

File: Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift [hunk 1]
```diff
@@ -122,6 +125,7 @@ internal struct CreateProvisioningProfileCommand: ParsableCommand {
         case intermediaryAppleCertificates = "intermediaryAppleCertificates"
         case certificateSigningRequestSubject = "certificateSigningRequestSubject"
         case profileName = "profileName"
+        case autoRegenerate = "autoRegenerate"
```
Comment (tinder-maxwellelliott) [inline Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift:128]: Should this just be performed in the decode using a ValidationError https://apple.github.io/swift-argument-parser/documentation/argumentparser/validation/

File: Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift [hunk 2]
```diff
@@ -286,18 +295,39 @@ internal struct CreateProvisioningProfileCommand: ParsableCommand {
...
     internal func run() throws {
-        let privateKey: Path = .init(privateKeyPath)
-        let csr: Path = try createCSR(privateKey: privateKey)
         let jsonWebToken: String = try jsonWebTokenService.createToken(
             keyIdentifier: keyIdentifier,
             issuerID: issuerID,
             secretKey: try files.read(Path(itunesConnectKeyPath))
         )
+        let deviceIDs: Set<String> = try iTunesConnectService.fetchITCDeviceIDs(jsonWebToken: jsonWebToken)
+        if autoRegenerate, profileName == nil {
```
Comment (tinder-maxwellelliott) [inline Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift:310]: Should this just be performed in a validate function? https://apple.github.io/swift-argument-parser/documentation/argumentparser/validation/

File: Sources/SignHereLibrary/Models/CreateProfileResponse.swift [hunk 3]
```diff
@@ -8,20 +8,5 @@
...
-            var expirationDate: Date
-        }
-        var id: String
-        var type: String
-        var attributes: Attributes
-    }
-    var data: CreateProfileResponseData
-}
+    var data: ProfileResponseData
+}
```
Comment (tinder-maxwellelliott) [inline Sources/SignHereLibrary/Models/CreateProfileResponse.swift:12]:
```suggestion
}

```

File: Sources/SignHereLibrary/Models/ProfileResponseData.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+            var data: [Data]
+        }
+        var devices: Devices
+    }
+    var id: String
+    var type: String
+    var attributes: Attributes
+    var relationships: Relationships
+}
```
Comment (tinder-maxwellelliott) [inline Sources/SignHereLibrary/Models/ProfileResponseData.swift:36]:
```suggestion
}

```

File: Sources/SignHereLibrary/Models/ProfileType.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    }
+
+    var usesDevices: Bool {
+        switch self {
+            case .appStore: return false
+            default: return true
+        }
+    }
+}
```
Comment (tinder-maxwellelliott) [inline Sources/SignHereLibrary/Models/ProfileType.swift:35]:
```suggestion
}

```

**Review by tinder-maxwellelliott — COMMENTED** [review]
Looking great

File: Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift (hunk 2 above)
Comment (omarzl) [reply Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift]: sure! done

**Pull request merged**